assignment-client/src/entities/EntityTreeSendThread.cpp

```diff
@@ -92,10 +93,14 @@ void EntityTreeSendThread::traverseTreeAndSendContents(SharedNodePointer node, O
             ViewFrustum viewFrustum;
             nodeData->copyCurrentViewFrustum(viewFrustum);
             EntityTreeElementPointer root = std::dynamic_pointer_cast<EntityTreeElement>(_myServer->getOctree()->getRoot());
-            int32_t lodLevelOffset = nodeData->getBoundaryLevelAdjust() + (viewFrustumChanged ? LOW_RES_MOVING_ADJUST : NO_BOUNDARY_ADJUST);
             startNewTraversal(viewFrustum, root, lodLevelOffset);
         }
     }
+
+    // If the previous traversal didn't finish, we'll need to resort the entities still in _sendQueue after calling traverse
+    EntityPriorityQueue prevSendQueue;
+    _sendQueue.swap(prevSendQueue);
+
```

Comment:
We only want to re-sort when the view actually changes AND the _sendQueue is not empty.  Move this swap and the actual re-sort inside the `if (viewFrustumChanged || repeatTraversal)` check above, but after the `startNewTraversal()` call since the new `_currentView` is updated therein and we would want it up-to-date before doing the re-sort.

You can then move the `lodLevelOffset` calculation back into its original context.